java/client/src/main/java/glide/connectors/handlers/CallbackDispatcher.java

```diff
@@ -107,11 +108,8 @@ public void completeRequest(Response response) {
             }
             future.completeAsync(() -> response);
         } else {
-            // TODO: log an error thru logger.
             // probably a response was received after shutdown or `registerRequest` call was missing
-            System.err.printf(
-                    "Received a response for not registered callback id %d, request error = %s%n",
-                    callbackId, response.getRequestError());
+            Logger.log(Logger.Level.ERROR, "callback dispatcher", "Received a response for not registered callback id " + callbackId + ", request error = " + response.getRequestError());
```

Comment:
you could import Logger.Level.ERROR
```suggestion
            Logger.log(ERROR, "callback dispatcher", "Received a response for not registered callback id " + callbackId + ", request error = " + response.getRequestError());
```